runtime/src/main/scala/akka/grpc/internal/InternalChannel.scala

```diff
@@ -5,14 +5,35 @@
 package akka.grpc.internal
 
 import scala.concurrent.Future
-
-import akka.Done
+import akka.{ Done, NotUsed }
 import akka.annotation.InternalApi
-
-import io.grpc.ManagedChannel
+import akka.grpc.{ GrpcResponseMetadata, GrpcSingleResponse }
+import akka.stream.scaladsl.Source
+import io.grpc.{ CallOptions, MethodDescriptor }
 
 /**
  * INTERNAL API
  */
 @InternalApi
-case class InternalChannel(managedChannel: ManagedChannel, done: Future[Done])
+abstract class InternalChannel {
```

Comment:
Nice seemed simple enough to get away from `ManagedChannel`. Do we also want to get away from `CallOptions` and `MethodDescriptor` at some point?

Reply:
> Do we also want to get away from `CallOptions` and `MethodDescriptor` at some point?

I don't mind those so much, they seem fairly reasonable.